ethereum/contracts/ERC20Vault.sol

```diff
@@ -0,0 +1,49 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.9;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+// Holds ETH on behalf of ETHApp
+contract ERC20Vault is Ownable {
+    using SafeERC20 for IERC20;
+
+    event Deposit(address account, address sender, uint256 amount);
+    event Withdraw(address account, address recipient, uint256 amount);
+
+    mapping(address => uint256) public balances;
+
+    receive() external payable {
+        revert("Must use deposit function");
+    }
+
+    // Accepts ETH from the caller.
```

Comment:
Please use the [NATSPEC](https://docs.soliditylang.org/en/v0.8.17/natspec-format.html) style comments. See https://github.com/Snowfork/snowbridge/blob/main/ethereum/contracts/BeefyClient.sol for an example.

Reply:
Addressed in https://github.com/Snowfork/snowbridge/pull/706/commits/4f7524253c72c6ccf7f518245e67b4d91caff153.